frontend/src/MapPath.js

```diff
@@ -1,16 +1,49 @@
 import React from "react";
 import PropTypes from "prop-types";
+import * as d3 from "d3";
 
 export class MapPath extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            fill: this.props.fill,
+        };
+        this.path_ref = React.createRef();
+    }
+
+    shouldComponentUpdate(nextProps) {
```

Comment:
THANK YOU for this excellent documentation about how this works -- will be very helpful for everyone else working with these transitions.